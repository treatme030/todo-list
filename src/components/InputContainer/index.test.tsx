import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import "@testing-library/jest-dom/extend-expect";
import 'jest-styled-components';
import { InputContainer } from './index';
import { ToDoListProvider } from 'contexts';

describe('<InputContainer />', () => {
  it('renders component correctly', () => {
    const { container } = render(<InputContainer />);

    const input = screen.getByPlaceholderText('할 일을 입력해 주세요');
    expect(input).toBeInTheDocument();
    const addButton = screen.getByText('추가');
    expect(addButton).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
  it('empties data after adding data', () => {
    render(<InputContainer />);

    const input = screen.getByPlaceholderText('할 일을 입력해 주세요') as HTMLInputElement;
    const addButton = screen.getByText('추가');
    expect(input.value).toBe('');

    userEvent.type(input, 'study react 1');
    expect(input.value).toBe('study react 1');
    userEvent.click(addButton);
    expect(input.value).toBe('');
  });
  it('adds input data to localStorage via Context', () => {
    render(
      <ToDoListProvider>
        <InputContainer />
      </ToDoListProvider>
    );

    const input = screen.getByPlaceholderText('할 일을 입력해 주세요');
    const addButton = screen.getByText('추가');
    expect(localStorage.getItem('ToDoList')).toBeNull();

    userEvent.type(input, 'study react 1');
    userEvent.click(addButton);
    expect(localStorage.getItem('ToDoList')).toBe('["study react 1"]');
  });
  it('calls the onAdd function when the user clicks Add button', () => {
    const handleClick = jest.fn();
    render(<InputContainer onAdd={handleClick} />);

    const input = screen.getByPlaceholderText('할 일을 입력해 주세요');
    const addButton = screen.getByText('추가');
    expect(handleClick).toHaveBeenCalledTimes(0);

    userEvent.click(addButton);
    expect(handleClick).toHaveBeenCalledTimes(0);

    userEvent.type(input, 'study react 1');
    userEvent.click(addButton);
    expect(handleClick).toHaveBeenCalledTimes(1);
  })
});

