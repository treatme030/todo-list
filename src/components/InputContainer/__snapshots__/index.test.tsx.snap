// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputContainer /> renders component correctly 1`] = `
.c2 {
  text-align: center;
  background-color: #304ffe;
  padding: 10px 20px;
  border-radius: 8px;
  cursor: pointer;
}

.c2:hover {
  background-color: #1e40fe;
}

.c2:active {
  box-shadow: inset 5px 5px 10px rgba(0,0,0,0.2);
}

.c3 {
  color: #fff;
  font-size: 16px;
}

.c1 {
  font-size: 16px;
  padding: 10px 10px;
  border-radius: 8px;
  border: 1px solid #bdbdbd;
  outline: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div>
  <div
    class="c0"
  >
    <input
      class="c1"
      placeholder="할 일을 입력해 주세요"
      value=""
    />
    <button
      class="c2"
    >
      <label
        class="c3"
      >
        추가
      </label>
    </button>
  </div>
</div>
`;
